modules/concurrency/include/hephaestus/concurrency/spinner.h

```diff
@@ -17,18 +17,21 @@ namespace heph::concurrency {
 /// the given fixed rate.
 class Spinner {
 public:
-  enum class SpinResult : bool { CONTINUE, STOP };
-  using StoppableCallback = std::function<SpinResult()>;
-  using Callback = std::function<void()>;
+  using TransitionCallback = std::function<void()>;
+  using PolicyCallback = std::function<bool()>;
 
-  /// @brief Create a spinner with a stoppable callback. A stoppable callback is a function that returns
-  /// SpinResult::STOP to indicate that the spinner should stop.
-  /// Example: a callback that stops after 10 iterations.
-  explicit Spinner(StoppableCallback&& stoppable_callback, double rate_hz = 0);
+  struct Callbacks {
+    TransitionCallback init_cb = []() {};
+    TransitionCallback spin_once_cb = []() {};
```

Comment:
I think we can still support the stoppable callback with this design

Reply:
We do, we just change the approach. Instead of injecting it into `spin()` via the executable callback, it's now a first class citizen via `shall_stop_cb`.